### PR TITLE
New package: boot-66serv-2.1.0

### PR DESCRIPTION
**Pull request by teldra**
based on https://github.com/void-linux/void-packages/pull/21142

### CODE REVIEW

File: srcpkgs/66-boot/template [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+short_desc="Stage 1 boot for 66"
+maintainer="teldra <teldra@rotce.de>"
+license="ISC"
+homepage="https://framagit.org/Obarun/boot-66serv"
+distfiles="https://framagit.org/Obarun/${_realpkgname}/-/archive/v${version}/${_realpkgname}-v${version}.tar.gz"
+checksum=6cf417a7eb5bcac152480f9a76711b06289ae672d1da7d36dc8186a6eef04235
+conf_files="/etc/66/rc.local"
+
+post_install() {
+	vsed -i "s/@vers@/${version}/" ${FILESDIR}/tty@
```
Comment (bougyman) [inline srcpkgs/66-boot/template:20]: I don't see @vers@ in tty@

Comment (teldra) [reply srcpkgs/66-boot/template:20]: Ah, this slipped through. :(

File: srcpkgs/66-boot/INSTALL.msg [hunk 2]
```diff
@@ -0,0 +1 @@
+To boot with 66, simply add the argument `init=/usr/bin/66' to your kernel commandline.
```
Comment (Duncaen) [inline srcpkgs/66-boot/INSTALL.msg:1]: This shouldn't be a INSTALL message.

File: srcpkgs/66-boot/INSTALL [hunk 3]
```diff
@@ -0,0 +1,10 @@
+if [ "$UPDATE" = "no" ]; then
+	case "${ACTION}" in
+	post)
+		66-tree -n boot
+		66-enable -t boot boot@system
+		66-tree -ncE default
+		66-enable -t default tty@tty1 tty@tty2 tty@tty3 tty@tty4
```
Comment (Duncaen) [inline srcpkgs/66-boot/INSTALL:7]: Not sure if its a good idea to do all of this in INSTALL, also wouldn't work right when doing install with `xbps-install --root`.

File: srcpkgs/66-boot/REMOVE [hunk 4]
```diff
@@ -0,0 +1,6 @@
+case "${ACTION}" in
+	post)
+		66-tree -R boot
+		66-tree -R default
```
Comment (Duncaen) [inline srcpkgs/66-boot/REMOVE:4]: I don't think this should be done in REMOVE, I assume this removes user data?

File: srcpkgs/66-boot/template [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Template file for '66-boot'
+pkgname=66-boot
+_realpkgname=boot-66serv
```
Comment (Duncaen) [inline srcpkgs/66-boot/template:3]: Why not use the real name?

File: srcpkgs/66-boot/INSTALL.msg (hunk 2 above)
Comment (teldra) [reply srcpkgs/66-boot/INSTALL.msg]: I will create a doc for docs.voidlinux.org

File: srcpkgs/66-boot/INSTALL (hunk 3 above)
Comment (teldra) [reply srcpkgs/66-boot/INSTALL]: I will create a doc for docs.voidlinux.org

File: srcpkgs/66-boot/template (hunk 5 above)
Comment (teldra) [reply srcpkgs/66-boot/template]: Yes, why not :)
Renamed.

File: srcpkgs/66-boot/REMOVE (hunk 4 above)
Comment (teldra) [reply srcpkgs/66-boot/REMOVE]: Yes, it removes userdata. Removed.

**Comment by flexibeast** [comment]
i'm interested in moving my Void system over to using 66, so i've been playing around with 66 on Void in a VM. Building `boot-66serv` from the template in this PR, i've got it working; my next step is to use @mobinmob's [void-66-packages](https://github.com/mobinmob/void-66-services/) for some of the services i need, and to hopefully contribute some new services to that repo (a wpa_supplicant service in particular).

i'm actually very new to s6 and 66, so i'm taking this opportunity to document things as i go, in the form of a putative section of the Handbook:

https://github.com/flexibeast/void-docs/blob/add-s6-and-66/src/config/services/s6-and-66/index.md

Please feel free to let me know of any errors i've made thus far, and to make suggestions about what else should/could be added!

**Comment by teldra** [comment]
> i'm interested in moving my Void system over to using 66, so i've been playing around with 66 on Void in a VM. Building `boot-66serv` from the template in this PR, i've got it working; my next step is to use @mobinmob's [void-66-packages](https://github.com/mobinmob/void-66-services/) for some of the services i need, and to hopefully contribute some new services to that repo (a wpa_supplicant service in particular).

With this template, make sure, you take the [v0.0.4.x branch](https://github.com/mobinmob/void-66-services/tree/v0.0.4.x) and even this is not completely tested. (and feel free to pr to this branch). ;)
And watch out for `make` file in root folder. ;)

> i'm actually very new to s6 and 66, so i'm taking this opportunity to document things as i go, in the form of a putative section of the Handbook:
> 
> https://github.com/flexibeast/void-docs/blob/add-s6-and-66/src/config/services/s6-and-66/index.md

We are also working on [void-docs](https://github.com/teldra/void-docs/tree/66-docs). (again: feel free to PR to branch). If you compare this to yours, there is indeed one mistake: You don't have to enable tty* in the tree you are calling root, because that should be started in your boot tree. You can cofigure the amount off ttys in the step [Configure](https://github.com/teldra/void-docs/blob/66-docs/src/config/services/66/index.md#configure)
The following aren't mistakes but small inconveniences:
You can add `init=/usr/bin/66` to your kernelcmdline and don't have to replace /usr/bin/init.
Your suggestions of packages to install can be reduced to only install `boot-66serv` which should pull needed dependencies.
Replacing `power`, `reboot` etc. can be more elegant.

As long as it is not merged, watch out for this https://github.com/void-linux/void-packages/pull/23330 PR, because 66 in repos has a small error, which creates folder '$datarootdir', where you invoke 66-tree.

I am glad, that this PR works for you! :)

**Comment by flexibeast** [comment]
> With this template, make sure, you take the v0.0.4.x branch and even this is not completely tested. (and feel free to pr to this branch). ;)
> And watch out for make file in root folder. ;)

Ah, thank you. :-)

> We are also working on void-docs.

Oh wow, great! Well, there's no need for my branch, then, so i'll remove it and work on your repo instead. To that end, i can submit PRs for ensuring the docs follow the Handbook style guide.

i presume other members of the Void team have already given their okay for this information being included in the Handbook? Just need to check, as i'm one of the people who would be reviewing and/or merging such an inclusion. :-)

Thanks for all your work!

**Comment by mobinmob** [comment]
@flexibeast : Thank you for taking an interest in our experiments - they run our machines, so they are halfway to world domination :P
I have fallen behind in testing all the changes in the 0.0.4.x branch - Ι want to merge it ASAP but I would like to keep everything in master working as we do not yet have a release. 
The doc changes are right now just enough to test 66 and boot-66serv using runit services for anything outside of boot-66serv. They work, but they are neither the end goal nor without problems. Replacing  the reboot/halt etc works but it is ugly - those files belong to a package and any change will be overwritten when the package is updated.
_void-66-services_, are very much a WIP right now. boot-66serv works reliably but there are some problems with configuration - mentioned above.

**Comment by flexibeast** [comment]
@mobinmob: You're welcome! Thank you to both you and @teldra for working on this, and for letting me know the state of play - i hope to find time to help with getting more things working, if i can. :-)

**Comment by mobinmob** [comment]
@teldra : Is tty@ needed? tty-rc@ seems to be used for ttys and it is included already.

**Comment by teldra** [comment]
Currently tty@ is needed, because it is pulled in in `/usr/share/66/service/boot@`.

**Comment by mobinmob** [comment]
@teldra : 2.2.0 is out. 
It includes documentation for boot@ and needs `lowdown` in the hostmakedepends array and `install`, `install-man` and `install-html` in the make_install_target array.

**Comment by teldra** [comment]
@mobinmob I am not interested in continuing this. Maybe you want to, so i will leave this open, until you say, if you want to continue or not.

**Comment by mobinmob** [comment]
> @mobinmob I am not interested in continuing this. Maybe you want to, so i will leave this open, until you say, if you want to continue or not.

Thank you for creating the packagins bits and maintaining them so far. I will gladly continue the work as I am using it, please do not delete your branch. 
I will post a new PR once there is an upstream update or I have finished some relevant work - whatever comes first!

**Comment by teldra** [comment]
@mobinmob There is an update for this package.

**Pull request closed without merging**